File: app/src/main/java/com/example/carlauncher/data/VehicleDataSourceFactory.java
package com.example.carlauncher.data;

import android.content.Context;

import com.example.carlauncher.data.mock.MockVehicleDataSource;
import com.example.carlauncher.data.vhal.VhalVehicleDataSource;

public final class VehicleDataSourceFactory {
    private VehicleDataSourceFactory() {
    }

    public static VehicleDataSource create(
            Context context,
            SourceType sourceType
    ) {
        if (sourceType == SourceType.VHAL) {
            return new VhalVehicleDataSource(context);
        }
        return new MockVehicleDataSource();
    }
}

File: app/src/main/java/com/example/carlauncher/data/VehicleDataSourceFactory.kt
package com.example.carlauncher.data

import android.content.Context
import com.example.carlauncher.data.mock.MockVehicleDataSource
import com.example.carlauncher.data.vhal.VhalVehicleDataSource

class VehicleDataSourceFactory private constructor() {
    companion object {

        @JvmStatic
        fun create(
            context: Context?,
            sourceType: SourceType?,
        ): VehicleDataSource? {
            if (sourceType == SourceType.VHAL) {
                return VhalVehicleDataSource(context)
            }
            return MockVehicleDataSource()
        }
    }
}
